<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitXMLReporter -->
<testsuite ignored="0" hostname="Piyushs-MacBook-Air.local" failures="2" tests="6" name="Surefire test" time="2.839" errors="0" timestamp="2024-03-18T16:25:35 IST">
  <testcase classname="com.test.WeatherTest" name="testFetchWeatherData" time="1.095"/>
  <testcase classname="com.test.WeatherTest" name="testTemperature" time="0.271"/>
  <testcase classname="com.test.WeatherTest" name="testHumidity" time="0.242"/>
  <testcase classname="com.test.WeatherTest" name="testFetchWeatherData_ValidCity" time="0.248"/>
  <testcase classname="com.test.WeatherTest" name="testFetchWeatherData_InvalidCity" time="0.7">
    <failure type="io.restassured.path.json.exception.JsonPathException" message="Failed to parse the JSON document">
      <![CDATA[io.restassured.path.json.exception.JsonPathException: Failed to parse the JSON document
at io.restassured.path.json.JsonPath$ExceptionCatcher.invoke(JsonPath.java:1002)
at io.restassured.path.json.JsonPath$4.doParseWith(JsonPath.java:967)
at io.restassured.path.json.JsonPath$JsonParser.parseWith(JsonPath.java:1047)
at io.restassured.path.json.JsonPath.get(JsonPath.java:202)
at io.restassured.path.json.JsonPath.getString(JsonPath.java:352)
at com.test.WeatherTest.testFetchWeatherData_InvalidCity(WeatherTest.java:69)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: groovy.json.JsonException: Lexing failed on line: 1, column: 1, while reading 'w', no possible valid JSON value or punctuation could be recognized.
at groovy.json.JsonLexer.nextToken(JsonLexer.java:86)
at groovy.json.JsonLexer$nextToken.call(Unknown Source)
at io.restassured.internal.path.json.ConfigurableJsonSlurper.parse(ConfigurableJsonSlurper.groovy:97)
at io.restassured.internal.path.json.ConfigurableJsonSlurper$parse.callCurrent(Unknown Source)
at io.restassured.internal.path.json.ConfigurableJsonSlurper.parseText(ConfigurableJsonSlurper.groovy:83)
at io.restassured.path.json.JsonPath$4$1.method(JsonPath.java:965)
at io.restassured.path.json.JsonPath$ExceptionCatcher.invoke(JsonPath.java:1000)
... 38 more
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- testFetchWeatherData_InvalidCity -->
  <testcase classname="com.test.WeatherTest" name="testFetchWeatherData_EmptyCity" time="0.215">
    <failure type="java.lang.IllegalArgumentException" message="Schema to use cannot be null">
      <![CDATA[java.lang.IllegalArgumentException: Schema to use cannot be null
at io.restassured.module.jsv.JsonSchemaValidator.validateSchemaIsNotNull(JsonSchemaValidator.java:270)
at io.restassured.module.jsv.JsonSchemaValidator.access$300(JsonSchemaValidator.java:75)
at io.restassured.module.jsv.JsonSchemaValidator$JsonSchemaValidatorFactory.create(JsonSchemaValidator.java:281)
at io.restassured.module.jsv.JsonSchemaValidator.matchesJsonSchema(JsonSchemaValidator.java:166)
at io.restassured.module.jsv.JsonSchemaValidator.matchesJsonSchemaInClasspath(JsonSchemaValidator.java:117)
at com.test.WeatherTest.testFetchWeatherData_EmptyCity(WeatherTest.java:78)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
... Removed 32 stack frames]]>
    </failure>
  </testcase> <!-- testFetchWeatherData_EmptyCity -->
</testsuite> <!-- Surefire test -->
